test/hooks/post-api-plugin/plugin-verification.js.test.js

```diff
@@ -0,0 +1,81 @@
+const pluginFunc = require('../../../src/hooks/post-api-plugin/plugin-verification');
+const { expect, test } = require('@twilio/cli-test');
+
+const CLI_NAME = 'twilio-cli';
+
+const getCliPlugin = () => ({
+  name: CLI_NAME,
+  commands: [{
+    id: 'login',
+    aliases: ['logon']
+  }, {
+    id: 'feedback'
+  }, {
+    id: 'api'
+  }]
+});
+
+const getApiPlugin = () => ({
+  name: 'api-plugin',
+  commands: [{
+    id: 'api'
+  }]
+});
+
+const getConformingPlugin = () => ({
+  name: 'plugin-meals',
+  commands: [{
+    id: 'breakfast'
+  }, {
+    id: 'lunch',
+    aliases: ['dinner']
+  }]
+});
+
+const getConflictingPlugin = () => ({
+  name: 'plugin-schmlugin',
+  commands: [{
+    id: 'logon'
+  }, {
+    id: 'feed:me',
+    aliases: ['feedback']
+  }]
+});
+
+const getFakeConfig = () => ({
+  name: CLI_NAME,
+  plugins: [getCliPlugin(), getApiPlugin()]
+});
+
+describe('hooks', () => {
+  describe('post-api-plugin', () => {
+    describe('plugin-verification', () => {
+      test.stderr().it('outputs nothing for just core plugins', ctx => {
+        ctx.config = getFakeConfig();
+
+        pluginFunc.call(ctx, getApiPlugin());
+
+        expect(ctx.stderr).to.be.empty;
+      });
+
+      test.stderr().it('outputs nothing if no conflicts in installed plugins', ctx => {
+        ctx.config = getFakeConfig();
+        ctx.config.plugins.push(getConformingPlugin());
+
+        pluginFunc.call(ctx, getApiPlugin());
+
+        expect(ctx.stderr).to.be.empty;
+      });
+
+      test.stderr().it('warns for each conflicting command', ctx => {
+        ctx.config = getFakeConfig();
+        ctx.config.plugins.push(getConflictingPlugin());
+
+        pluginFunc.call(ctx, getApiPlugin());
+
+        expect(ctx.stderr).to.contain('logon');
+        expect(ctx.stderr).to.contain('feedback');
+      });
+    });
+  });
+});
```

Comment:
Is it worth creating actual sample plugins that would error for a sanity check? Or have you already done that?

Otherwise, the tests LGTM.

Reply:
I'm okay with that, but think it falls outside the scope of this PR.

This test illustrates that the verification logic is sound, but it doesn't verify that it actually gets invoked during command execution (i.e., there's no test to verify our API plugin is properly configured, that it executes the proper hook, and that the plugin verification is configured properly).

This falls in the realm of integration tests, which we don't currently have: installing the CLI, running commands, updating it, and doing this on all supported operating systems.

Reply:
That sounds good. I ran into a similar issue with the welcome message.